src/Chapter3.hs

```diff
@@ -971,6 +1079,21 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Mon | Tue | Wed | Thu | Fri | Sat | Sun deriving (Show, Eq, Ord, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend day = day >= Sat
+
+nextDay :: Weekday -> Weekday
+nextDay day
+    | day == Sun = Mon
+    | otherwise = succ day
+
+daysToParty :: Weekday -> Int
+daysToParty day
+    | day == Fri = 0
+    | otherwise = 1 + daysToParty (nextDay day)
```

Comment:
Interesting solution reusing the `nextDay` function! Though, you can implement it more efficiently by using the `fromEnum` method of the `Enum` typeclass